print_d.c

```diff
@@ -2,37 +2,34 @@
 #include <stdio.h>
 #include "main.h"
 
-/* printf_integer_recursive - prints a single or multi digit integer
- * @num: a single or multi digit integer
- * Return: The length of the integer
+/* printf_integer - prints a singe or multi digit integer
+ * @val: variable length argument list
+ * Return: the length of the integer
  */
-int print_integer_recursive(int num)
+int printf_integer(va_list val)
 {
+	int num = va_arg(val, int);
+	char buffer[20];
+	int len = 0, digitCount = 0;
+
 	if (num < 0)
 	{
 		_putchar('-');
 		num = -num;
+		digitCount++;
 	}
 
-	if (num / 10)
+	while (num)
 	{
-		int len = print_integer_recursive(num / 10);
-		_putchar(num % 10 + '0');
-		return len + 1;
+		buffer[len++] = num % 10 + '0';
+		num /= 10;
+		digitCount++;
 	}
-	else
+
+	// Print the formatted string
+	for (int i = len - 1; i >= 0; i--)
 	{
-		_putchar(num % 10 + '0');
-		return 1;
+		_putchar(buffer[i]);
```

Comment:
Thank you for the explanation, @zonafrank.  This is where you implement the iterate in reverse method.
In future projects, I think it would be good if we chat and share knowledge/ideas here. 
Good practice and enjoyable. At least to me.